lsp-mode.el

```diff
@@ -5224,10 +5279,27 @@ It will show up only if current point has signature help."
       (funcall action-handler action)
     (lsp--send-execute-command command arguments?)))
 
+(defun lsp--select-action-enabled-or-die (actions)
+  "Prompt the user to select an action from ACTIONS.
+Even disabled code actions are shown. If such an action is
+chosen, `user-error'."
+  (let ((action (lsp--select-action actions)))
+    (-when-let ((&CodeAction :disabled?) action)
+      (user-error "%s is disabled: %s" (lsp:code-action-title action)
+                  (lsp:code-action-disabled-reason disabled?)))
+    action))
+
 (lsp-defun lsp-execute-code-action ((action &as &CodeAction :command? :edit?))
   "Execute code action ACTION.
-If ACTION is not set it will be selected from `lsp-code-actions-at-point'."
-  (interactive (list (lsp--select-action (lsp-code-actions-at-point))))
+Interactively, ask for a code action. If the prefix argument is
+unspecified, the set of code actions will be filtered to only
+include those that are not disabled."
+  (interactive
+   (list
+    (funcall (if current-prefix-arg
+                 #'lsp--select-action-enabled-or-die
+               #'lsp--select-enabled-action)
+             (lsp-code-actions-at-point))))
```

Comment:
seems like if you drop lsp--select-enabled-action/lsp--select-action-enabled-or-die and pass an optional parameter to lsp-code-actions-at-point the code will be simplified/shortened a lot.

Reply:
Fixed.